### ISSUE TITLE
System media collections are not created when cache is not cleared manually

### ISSUE THREAD

**Issue by plozmun** [sulu/sulu#5245]
| Q | A
| --- | ---
| Sulu Version | 2.0.6
| Browser Version | Chrome last 

# Actual Behavior

Hi, i have a strange issue. In a fresh install, System media collections are not created, all media is empty. Then if i try to upload an image in a organization it fails becase it trying to upload the image in collection "6" that not exists. 
It's strange because I recently saw that they were created correctly when trying to upload an image 

# Expected Behavior

System collections must be created when install and maybe protected.

# Steps to Reproduce

Clear db, execute install and inspect media. Later try to add an image to a organization

**Comment by alexander-schranz** [comment]
Currently if you did `bin/console sulu:build dev --destroy` you need to clear the cache manually else the system collection id mapping are still in cache. We removed the cache:clear some time ago in the sulu:build command (#3671) because symfony did not longer allow todo anything after it.

**Comment by danrot** [comment]
I thought we already have an issue like that, where was also described that we actually should not store this information in the cache... But I can't find it anymore :confused: The only thing I've found with some information is #374, which talks about a similar issue for categories, and I've added some comments about this.

So I think we are going to leave that issue open, but I am going to rename it.

**Comment by plozmun** [comment]
Debugging a little this is the exception that is launched https://github.com/sulu/sulu/blob/release/2.0/src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php#L643 but it's not catched anywhere.
Where is the "system collection" supposed to be created? If I can help.

**Comment by danrot** [comment]
I think the problem is a bit harder to solve... We are currently [storing all system collections in the symfony cache](https://github.com/sulu/sulu/blob/b48343f/src/Sulu/Component/Media/SystemCollections/SystemCollectionManager.php#L105). And since you are not deleting the cache it will still take these values, leading to this strange behavior.

The cache is also required in order to create the collections only once in the database. I think we need to somehow remove these things from the cache, but I am not totally sure right now about all other parts that would be impacted by that. I mean definitely everything in the `Sulu\Component\Media\SystemCollections` namespace. The other things that would still have to work are:

1. Initial creation of the collections in the database
2. Create these collections only once, no matter of the circumstances

If you would have an idea how to solve that we can discuss that in here, and we would happily accept a PR :slightly_smiling_face:

**Comment by alexander-schranz** [comment]
At current state I think the simpliest solution would be using the `sulu_media.system_collections.cache` service which could just be cleared in the `sulu:build` creating an own [SuluBuilder](https://github.com/sulu/sulu/blob/2.0.6/src/Sulu/Bundle/SecurityBundle/Build/UserBuilder.php) service so.

**Comment by danrot** [comment]
Right, that's something that could work... There will probably still be other problems not being solved by this approach, but it would probably get rid of this error for most of the users.

@plozmun Would you like to try that?

**Comment by plozmun** [comment]
I think that DataCache ( which is a filesystem cache ) should be changed by 'cache.app' . Now it store data in a file called 'system_collection.cache' with this info:
```
a:6:{s:4:"root";i:1;s:10:"sulu_media";i:2;s:24:"sulu_media.preview_image";i:3;s:12:"sulu_contact";i:4;s:20:"sulu_contact.contact";i:5;s:20:"sulu_contact.account";i:6;}
```

**Comment by danrot** [comment]
Well, for your error it does not matter how the cache is stored, you just have to delete it. Although it would probably make sense to move it to a Symfony Cache. Then the builder would probably also be easier to write, because we could make use of the Symfony commands for their cache.

**Comment by alexander-schranz** [comment]
Totally would make sense to remove the `Sulu\Component\Cache\DataCache` and replace it with the a cache pool in `cache.app`, but currently from backward compatibility I would not change that and just clear the cache in the builder.

**Comment by plozmun** [comment]
Ok! I tried to change it with AdapterInterface and it's not too mucho work, maybe in another PR.

**Comment by danrot** [comment]
Fixed in #5248
